pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -189,6 +189,12 @@ func (r *ReconcileSyncSetInstance) Reconcile(request reconcile.Request) (reconci
 		return r.syncDeletedSyncSetInstance(ssi, ssiLog)
 	}
 
+	// Assume applied is false until we successfully apply everything to cluster.
+	ssi.Status.Applied = false
+	original := ssi.DeepCopy()
+	// Always update status if anything has changed
+	defer r.updateSyncSetInstanceStatus(ssi, original, ssiLog)
```

Comment:
Scenario:
- A sync has applied a syncsetinstance successfully
- We run this code again, with 0 changes to the syncsetinstance
- The sync this time, though, fails to apply that syncsetinstance

Won't that set `.status.applied` to false, even though this syncsetinstance was successfully sync'd earlier?



Also, what happens if updateSyndSetInstanceStatus errors out?

Reply:
Good question, so yes, in this metric right now, it would almost be better worded as "applying", rather than "applied". It's an indicator that this syncset is currently working ok. If nothing has changed but for some reason we can't apply, it's going to start reporting an issue. I think this is probably desirable and ok. We can structure the alerts such that we need it to be failing for some period of time before it triggers I think, items might show up briefly if there's an issue and then clear within a few seconds or minutes on requeue.

Does the behavior seem ok? And if so should I rename to applying or some other term?

Reply:
I guess it depends on what we're trying to do. I'm fine with this behavior. Is this something that SRE is asking for?

If so, I think SRE would like to use this in order to know if the syncsetinstance has ever been applied successfully. I think "was the last sync successful" is important, but less meaningful to them (I could be wrong).

Assuming I'm wrong, I'd probably rename this to `status.lastSyncSucceeded` (very wordy) or maybe just `status.syncSucceeded`. If the behavior stays the same, I think `status.applied` is not an accurate name and will likely confuse people.

Reply:
I've been raising that all clusters have had broken syncsets for months to them, and offered to get them a little more info on what specific syncsets are busted in the metrics instead of just X clusters have something wrong.

I'm not wondering if maybe there's enough info available for me to add a label indicating if this instance has *ever* synced to the cluster or not. (I think we can safely skip the complexity of determining if the current hash has ever synced successfully, unless that's really easy) Will check with @abutcher